### PR TITLE
workflows/docs: Check for broken links using the existing `rake test` task

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

- We had `rake test` as a task that uses `html-proofer` to scan for broken links, but we never actually ran it in CI or anywhere before building the site.
- This will make things slower (or maybe it's taking ages because of my dodgy hotel internet connection), maybe we should only run it on a schedule as a separate job?
- Fixes https://github.com/Homebrew/brew/issues/15908.

### CODE REVIEW

**Comment by issyl0** [comment]
https://github.com/gjtorikian/html-proofer/blob/main/html-proofer.gemspec#L20

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -44,6 +44,29 @@ jobs:
...
+          test-bot: false
+
+      - name: Install Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.1"
+          bundler-cache: true
+          working-directory: ${{ steps.set-up-homebrew.outputs.repository-path }}/docs
+
+      - name: Check for broken links in docs
```
Comment (MikeMcQuaid) [inline .github/workflows/docs.yml:66]: Maybe worth rolling this in under `Build docs site` in `linting` as it's going to be building in here anyway and doesn't seem too slow?

File: docs/Rakefile [hunk 2]
```diff
@@ -19,17 +19,19 @@ task test: :build do
...
+    ignore_status_codes: [0, 403],
     assume_extension: true,
     check_favicon: true,
     check_opengraph: true,
     check_html: true,
     check_img_http: true,
     enforce_https: true,
-    file_ignore: [
+    check_external_hash: false,
+    check_internal_hash: false,
```
Comment (MikeMcQuaid) [inline docs/Rakefile:30]: Might be worth keeping these in and whitelisting e.g. GitHub.com usage. My settings, if they are useful: https://github.com/MikeMcQuaid/mikemcquaid.com/blob/master/Rakefile

File: .github/workflows/docs.yml (hunk 1 above)
Comment (issyl0) [reply .github/workflows/docs.yml:66]: HTML Proofer needs Ruby 3 and the rest of the `linting` job runs on Ruby 2.7. Do we mind if we bump that up?

Comment (MikeMcQuaid) [reply .github/workflows/docs.yml]: Go for it!

**Review by EricFromCanada — APPROVED** [review]

File: docs/Rakefile [hunk 3]
```diff
@@ -19,19 +19,23 @@ task test: :build do
...
-      "http://formulae.brew.sh",
-      %r{https://github.com/Homebrew/brew/edit/gh-pages//},
-    ]
+    ignore_urls: [
+      %r{https://formulae.brew.sh"},
+      %r{https://github.com/}
+    ],
+    cache: {
+      timeframe: {
+        external: "1h"
```
Comment (MikeMcQuaid) [inline docs/Rakefile:37]: @issyl0 does this setup the GitHub Actions cache across workflows? If not, might be a nice extension.

**Review by MikeMcQuaid — COMMENTED** [review]
Great work!

**Pull request merged**